src/main/update-app/download-update/download-update.injectable.ts

```diff
@@ -6,21 +6,31 @@ import { getInjectable } from "@ogre-tools/injectable";
 import downloadPlatformUpdateInjectable from "../download-platform-update/download-platform-update.injectable";
 import downloadingUpdateStateInjectable from "../../../common/application-update/downloading-update/downloading-update-state.injectable";
 import discoveredVersionStateInjectable from "../../../common/application-update/discovered-version/discovered-version-state.injectable";
-import { runInAction } from "mobx";
+import { action, runInAction } from "mobx";
+import progressOfUpdateDownloadInjectable from "../../../common/application-update/progress-of-update-download/progress-of-update-download.injectable";
 
 const downloadUpdateInjectable = getInjectable({
   id: "download-update",
+
   instantiate: (di) => {
     const downloadPlatformUpdate = di.inject(downloadPlatformUpdateInjectable);
     const downloadingUpdateState = di.inject(downloadingUpdateStateInjectable);
     const discoveredVersionState = di.inject(discoveredVersionStateInjectable);
+    const progressOfUpdateDownload = di.inject(progressOfUpdateDownloadInjectable);
+
+    const updateDownloadProgress = action((percentage: number) => {
+      progressOfUpdateDownload.set(percentage);
+    });
 
     return async () => {
       runInAction(() => {
+        progressOfUpdateDownload.set(0);
         downloadingUpdateState.set(true);
       });
 
-      const { downloadWasSuccessful } = await downloadPlatformUpdate();
+      const { downloadWasSuccessful } = await downloadPlatformUpdate(
```

Comment:
Does does `downloadPlatformUpdate` not inject it itself?

Reply:
No. Separation `downloadUpdate` and `downloadPlatformUpdate` is to keep electron out from business logic. `*Platform*` contains electron specific stuff which is overridden from behavioural unit tests.

Reply:
Yeah but it is an injectable that would be able to be overidden anyway.

Reply:
We want to override the electron part, not our business logic.

Reply:
Sure, but isn't the point of DI the ability to override arbitrary deep dependencies?

Reply:
These abstractions are used just for what you describe, so that the override happens in the "sweet spot", where we keep the "business logic", and mock an arbitrary framework (here Electron), being a technical detail.

Reply:
Resolved for being clarified?